auth_with_external_browser_test.go

```
@@ -0,0 +1,189 @@
+		cleanupBrowserProcesses()
+		cfg.ClientStoreTemporaryCredential = 1
+		conn, _ := createConnection(getDbHandler(cfg))
+		_, err := conn.QueryContext(context.Background(), "SELECT 1")
+		assertNilF(t, err, fmt.Sprintf("Failed to run a query. err: %v", err))
+	})
+
+	t.Run("Verify validation of IDToken if option disabled", func(t *testing.T) {
+		cleanupBrowserProcesses()
+		cfg.ClientStoreTemporaryCredential = 0
+		tOut := "authentication timed out"
+		_, err := createConnection(getDbHandler(cfg))
+		assertTrueF(t, err.Error() == tOut, fmt.Sprintf("Expected %v, but got %v", tOut, err))
+	})
+}
+
+type ExternalBrowserProcess struct {
+	Success string
+	Fail    string
+	Timeout string
+}
+
+var externalBrowserType = ExternalBrowserProcess{
+	Success: "success",
+	Fail:    "fail",
+	Timeout: "timeout",
+}
+
+func cleanupBrowserProcesses() {
+	const cleanBrowserProcessesPath = "/externalbrowser/cleanBrowserProcesses.js"
+	_, err := exec.Command("node", cleanBrowserProcessesPath).Output()
+	if err != nil {
+		log.Fatalf("failed to execute command: %v", err)
+	}
+}
+
+func provideCredentials(ExternalBrowserProcess string, user string, password string) {
+	const provideBrowserCredentialsPath = "/externalbrowser/provideBrowserCredentials.js"
+	_, err := exec.Command("node", provideBrowserCredentialsPath, ExternalBrowserProcess, user, password).Output()
+	if err != nil {
+		log.Fatalf("failed to execute command: %v", err)
+	}
+}
+
+func connectToSnowflake(cfg *Config, query string, isCatchException bool) (err error) {
+	dsn, err := DSN(cfg)
+	if err != nil {
+		log.Fatalf("failed to create DSN from Config: %v, err: %v", cfg, err)
+	}
+	rows, err := executeQuery(query, dsn)
+	if isCatchException && err != nil {
+		log.Fatalf("failed to run a query. %v, err: %v", query, err)
+	} else if err != nil {
+		return err
+	}
+	defer rows.Close()
+	var v int
+	var hasAnyRows bool
+	for rows.Next() {
+		hasAnyRows = true
+		err := rows.Scan(&v)
+		if isCatchException && err != nil {
```

What if `isCatchException` is false? We will silently swallow an error?